taxi/tests/test_views.py

```diff
@@ -0,0 +1,174 @@
+        self.assertEqual(res.status_code, 200)
+        self.assertEqual(
+            list(res.context["manufacturer_list"]),
+            list(manufacturers)
+        )
+
+
+class PrivateCarTest(TestCase):
+
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            "test",
+            "test12345",
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_cars(self):
+        manufacturer = Manufacturer.objects.create(
+            name="Toyota",
+            country="Japan",
+        )
+        Car.objects.create(
+            model="Corolla",
+            manufacturer=manufacturer
+        )
+        Car.objects.create(
+            model="Camry",
+            manufacturer=manufacturer
+        )
+
+        res = self.client.get(CAR_URL)
+        cars = Car.objects.all()
+        self.assertEqual(res.status_code, 200)
+        self.assertEqual(
+            list(res.context["car_list"]),
+            list(cars)
+        )
+
+
+class ToggleAssignToCarTestCase(TestCase):
+    def setUp(self):
+        self.client = Client()
+        self.manufacturer = Manufacturer.objects.create(
+            name="Toyota",
+            country="Japan"
+        )
+        self.driver = get_user_model().objects.create_user(
+            username="user",
+            password="user12345",
+            license_number="LOL12345"
+        )
+        self.car = Car.objects.create(
+            model="RAV4",
+            manufacturer=self.manufacturer
+        )
+        self.url = reverse(
+            "taxi:toggle-car-assign",
+            args=[self.car.id]
+        )
+
+    def test_toggle_assign_to_car(self):
+        self.client.force_login(self.driver)
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 302)
+        self.assertTrue(self.car in self.driver.cars.all())
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 302)
+        self.assertTrue(self.car not in self.driver.cars.all())
+
+
+class TestSearchForms(TestCase):
+    def test_driver_search_form_valid_data(self):
+        form = DriverSearchForm(data={"username": "test"})
+        self.assertTrue(form.is_valid())
+
+    def test_driver_search_form_no_data(self):
+        form = DriverSearchForm(data={})
+        self.assertTrue(form.is_valid())
+
+    def test_car_search_form_valid_data(self):
+        form = CarSearchForm(data={"model": "Toyota"})
+        self.assertTrue(form.is_valid())
+
+    def test_car_search_form_no_data(self):
+        form = CarSearchForm(data={})
+        self.assertTrue(form.is_valid())
+
+    def test_manufacturer_search_form_valid_data(self):
+        form = ManufacturerSearchForm(data={"name": "Toyota"})
+        self.assertTrue(form.is_valid())
+
+    def test_manufacturer_search_form_no_data(self):
+        form = ManufacturerSearchForm(data={})
+        self.assertTrue(form.is_valid())
```

Comment:
u are testing valid forms here, try to test this forms through ur views, to check context data from response